components/motor/gpio/motor_encoder_test.go

```diff
@@ -78,6 +80,114 @@ func MakeIncrementalBoard(t *testing.T) *fakeboard.Board {
 	return &b
 }
 
+func MakeControlConfig(t *testing.T) control.Config {
+	cfg := control.Config{
+		Blocks: []control.BlockConfig{
+			{
+				Name: "endpoint",
+				Type: "endpoint",
+				Attribute: utils.AttributeMap{
+					"motor_name": "MotorFake",
+				},
+				DependsOn: []string{"PID"},
+			},
+			{
+				Name: "trapz",
+				Type: "trapezoidalVelocityProfile",
+				Attribute: utils.AttributeMap{
+					"max_acc":    30000.0,
+					"max_vel":    4000.0,
+					"pos_window": 0.0,
+					"kpp_gain":   0.45,
+				},
+				DependsOn: []string{"set_point", "endpoint"},
+			},
+			{
+				Name: "sum",
+				Type: "sum",
+				Attribute: utils.AttributeMap{
+					"sum_string": "+-",
+				},
+				DependsOn: []string{"derivative", "trapz"},
+			},
+			{
+				Name: "set_point",
+				Type: "constant",
+				Attribute: utils.AttributeMap{
+					"constant_val": 0.0,
+				},
+				DependsOn: []string{},
+			},
+			{
+				Name: "derivative",
+				Type: "derivative",
+				Attribute: utils.AttributeMap{
+					"derive_type": "backward1st1",
+				},
+				DependsOn: []string{"endpoint"},
+			},
+			{
+				Name: "PID",
+				Type: "PID",
+				Attribute: utils.AttributeMap{
+					"tune_ssr_value": 2.0,
+					"kI":             0.53977,
+					"int_sat_lim_lo": -255.0,
+					"int_sat_lim_up": 255.0,
+					"limit_up":       255.0,
+					"tune_method":    "ziegerNicholsSomeOvershoot",
+					"limit_lo":       -255.0,
+					"tune_step_pct":  0.35,
+					"kD":             0.0,
+					"kP":             0.048401,
+				},
+				DependsOn: []string{"sum"},
+			},
+		},
+		Frequency: 100.0,
+	}
+	return cfg
+}
+
+func TestCreateMotorWithControls(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	controlCfg := MakeControlConfig(t)
+	cfg := Config{TicksPerRotation: 100, MaxRPM: 100, ControlLoop: controlCfg}
+	fakeMotor := &fakemotor.Motor{
+		MaxRPM:           100,
+		Logger:           logger,
+		TicksPerRotation: 100,
+		OpMgr:            operation.NewSingleOperationManager(),
+	}
+
+	ctx := context.Background()
+	b := MakeSingleBoard(t)
+	deps := make(resource.Dependencies)
+	deps[board.Named("main")] = b
+
+	ic := single.Config{
+		BoardName: "main",
+		Pins:      single.Pin{I: "10"},
+	}
+
+	rawcfg := resource.Config{Name: "enc1", ConvertedAttributes: &ic}
+	e, err := single.NewSingleEncoder(ctx, deps, rawcfg, logging.NewTestLogger(t))
```

Comment:
beautiful test, but I'd like to move away from relying on the exported functions of encoders at least. If not a bother to add it as a dependency through a config and a setup function, please do so!